idaes/core/surrogate/pysmo/sampling.py

```diff
@@ -594,6 +596,12 @@ def __init__(
             self.number_of_samples = number_of_samples
             self.x_data = bounds_array  # Only x data will be present in this case
 
+        if rand_seed is not None:
+            if not isinstance(rand_seed, int):
```

Comment:
Would it be better to do `int(rand_seed)` instead to catch cases where `rand_seed` is an integer in `float` form? You could either do this as a `try/except` or even just rely on the native exception from `int()`.

Reply:
Done.